File: todolist/src/types/finance.ts
// ─── 分类 ───
export interface TransactionCategory {
  id: number;
  name: string;
  icon: string;
  color: string;
  created_at: string;
}

export const FINANCE_CATEGORY_ICONS: Record<string, string> = {
  '学习': 'ReadOutlined',
  '餐饮': 'ShopOutlined',
  '购物': 'ShoppingCartOutlined',
  '交通': 'CarOutlined',
  '娱乐': 'SmileOutlined',
  '医疗': 'MedicineBoxOutlined',
  '服务': 'ToolOutlined',
  '转账': 'SwapOutlined',
  '借款': 'HandshakeOutlined',
  '红包': 'GiftOutlined',
  '生活缴费': 'ThunderboltOutlined',
  '其他': 'EllipsisOutlined',
};

// ─── 账户 ───
export interface Account {
  id: string;
  name: string;
  balance: number;
  is_savings: boolean;
  color?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateAccountParams {
  name: string;
  balance?: number;
  is_savings?: boolean;
  color?: string;
}

export interface UpdateAccountParams {
  name?: string;
  balance?: number;
  is_savings?: boolean;
  color?: string;
}

// ─── 交易 ───
export type TransactionType = 'expense' | 'income' | 'transfer_out' | 'transfer_in';

export interface Transaction {
  id: string;
  type: TransactionType;
  amount: number;
  category_id: number;
  account_id: string;
  target_account_id?: string;
  transfer_id?: string;
  date: string;
  note: string;
  created_at: string;
  updated_at: string;
}

export interface CreateTransactionParams {
  type: TransactionType;
  amount: number;
  category_id: number;
  account_id: string;
  target_account_id?: string;
  date: string;
  note?: string;
}

export interface UpdateTransactionParams {
  type?: TransactionType;
  amount?: number;
  category_id?: number;
  account_id?: string;
  date?: string;
  note?: string;
}

export interface TransactionFilter {
  year?: number;
  month?: number;
  category_id?: number;
  account_id?: string;
  type?: TransactionType;
  keyword?: string;
}

// ─── 统计 ───
export interface FinanceStats {
  monthly_expense: number;
  monthly_income: number;
  savings_total: number;
  net_worth: number;
  expense_count: number;
  income_count: number;
  savings_count: number;
}

export interface MonthlyChartData {
  month: string;
  expense: number;
  income: number;
}

export interface CategoryStat {
  category_id: number;
  category_name: string;
  category_color: string;
  total: number;
  count: number;
}

export interface DailyHeatmapCell {
  date: string;
  expense: number;
}

export interface NetWorthPoint {
  month: string;
  net_worth: number;
}

// ─── 用于 UI 展示的增强类型 ───
export interface TransactionDisplay extends Transaction {
  category_name: string;
  category_icon: string;
  category_color: string;
  account_name: string;
  target_account_name?: string;
}

File: todolist/src-tauri/src/models/finance.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TransactionCategory {
    pub id: i32,
    pub name: String,
    pub icon: String,
    pub color: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Account {
    pub id: String,
    pub name: String,
    pub balance: f64,
    pub is_savings: bool,
    pub color: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateAccountParams {
    pub name: String,
    #[serde(default)]
    pub balance: f64,
    #[serde(default)]
    pub is_savings: bool,
    pub color: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateAccountParams {
    pub name: Option<String>,
    pub balance: Option<f64>,
    pub is_savings: Option<bool>,
    pub color: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Transaction {
    pub id: String,
    #[serde(rename = "type")]
    pub type_: String,
    pub amount: f64,
    pub category_id: i32,
    pub account_id: String,
    pub target_account_id: Option<String>,
    pub transfer_id: Option<String>,
    pub date: String,
    pub note: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateTransactionParams {
    #[serde(rename = "type")]
    pub type_: String,
    pub amount: f64,
    pub category_id: i32,
    pub account_id: String,
    pub target_account_id: Option<String>,
    pub date: String,
    #[serde(default)]
    pub note: String,
}

#[derive(Debug, Deserialize)]
pub struct UpdateTransactionParams {
    #[serde(rename = "type")]
    pub type_: Option<String>,
    pub amount: Option<f64>,
    pub category_id: Option<i32>,
    pub account_id: Option<String>,
    pub date: Option<String>,
    pub note: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct TransactionFilter {
    pub year: Option<i32>,
    pub month: Option<i32>,
    pub category_id: Option<i32>,
    pub account_id: Option<String>,
    #[serde(rename = "type")]
    pub type_: Option<String>,
    pub keyword: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FinanceStats {
    pub monthly_expense: f64,
    pub monthly_income: f64,
    pub savings_total: f64,
    pub net_worth: f64,
    pub expense_count: i64,
    pub income_count: i64,
    pub savings_count: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MonthlyChartData {
    pub month: String,
    pub expense: f64,
    pub income: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CategoryStat {
    pub category_id: i32,
    pub category_name: String,
    pub category_color: String,
    pub total: f64,
    pub count: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DailyHeatmapCell {
    pub date: String,
    pub expense: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NetWorthPoint {
    pub month: String,
    pub net_worth: f64,
}
